docs/basics/101-127-procedures.rst

```diff
@@ -0,0 +1,361 @@
+      from datalad.utils import create_tree
+
+      ds = require_dataset(
+          sys.argv[1],
+          check_installed=True,
+          purpose='showcase an example procedure')
+
+      # this is the content for file "example"
+      content = """\
+      This file was created by a custom procedure! Neat, huh?
+      """
+
+      # create a directory structure template. Write
+      tmpl = {
+          'somedir': {
+              'example': content,
+          },
+          'example2': sys.argv[2] if sys.argv[2] else "got no input"
+      }
+
+      # actually create the structure in the dataset
+      create_tree(ds.path, tmpl)
+
+      # rule to store 'example' Git
+      ds.repo.set_gitattributes([('example', {'annex.largefiles': 'nothing'})])
+
+      # save the dataset modifications
+      ds.save(message="Apply custom procedure")
+
+      EOT
+
+   .. runrecord:: _examples/DL-101-127-105
+      :language: console
+      :workdir: procs/somedataset
+
+      $ datalad save -m "add custom procedure"
+
+   At this point, the dataset contains the custom procedure ``example``.
+   This is how it can be executed and what it does:
+
+   .. runrecord:: _examples/DL-101-127-106
+      :language: console
+      :workdir: procs/somedataset
+
+      $ datalad run-procedure example "this text will be in the file 'example2'"
+
+   .. runrecord:: _examples/DL-101-127-107
+      :language: console
+      :workdir: procs/somedataset
+
+      #the directory structure has been created
+      $ tree
+
+   .. runrecord:: _examples/DL-101-127-108
+      :workdir: procs/somedataset
+      :language: console
+
+      #lets check out the contents in the files
+      $ cat example2  && echo '' && cat somedir/example
+
+   .. todo::
+
+      At which point in this workflow do I include the help in the
+      config file?
+
+   .. runrecord:: _examples/DL-101-127-109
+      :workdir:  procs/somedataset
+      :language: console
+
+      $ git config -f .datalad/config datalad.procedures.example.help "A toy example"
+      $ datalad save -m "add help description"
+
+   To find out more about a given procedure, you can ask for help:
+
+   .. runrecord:: _examples/DL-101-127-110
+      :workdir: procs/somedataset
+      :language: console
+
+      $ datalad run-procedure --help-proc example
+
+   .. todo::
+
+      It might be helpful to have (or reference) a table with all available
+      procedures and a short explanation. Maybe on the cheatsheet.
+
+Summing up, DataLads :command:`run-procedure` command is a handy tool
+with useful existing procedures but much flexibility for own
+DIY procedure scripts. With the information of the last three sections
+you should be able to write and understand necessary configurations,
+but you can also rely on existing, preconfigured templates in the
+form of procedures, and even write and distribute your own.
+
+Therefore, envision procedures as
+configuration helpers that can minimize technical complexities
```

Comment:
Although the default procedures are all about configs, I would try to not too strongly emphasize the idea that procedures in general are mainly for configuration. The original idea emerged from a need to have structural templates for a dataset. But those kind of things are simply very use-case spcific, so it doesn't make sense to maintain them in datalad (core).